web/src/helpers/analytics.ts

```
@@ -20,21 +20,15 @@ import { DestinationTypeEnum } from 'Generated/schema';
 import mx from 'mixpanel-browser';
 import storage from 'Helpers/storage';
 import { ANALYTICS_CONSENT_STORAGE_KEY } from 'Source/constants';
-
-// TODO: Pending backend to work
-const mixpanelPublicToken = process.env.MIXPANEL_PUBLIC_TOKEN;
-
-const envCheck =
-  mixpanelPublicToken &&
-  // TODO: Pending backend to work
-  storage.local.read<boolean>(ANALYTICS_CONSENT_STORAGE_KEY) &&
-  process.env.NODE_ENV === 'production';
+import { AlertSummaryFull } from 'Source/graphql/fragments/AlertSummaryFull.generated';
 
 const evaluateTracking = (...args) => {
```

@3nvi @alexmylonas Just a minor note that I just spotted. 
In case `mixpanel` throws an error due to connection issues or any unknown errors this one will bubble up to the `ErrorBoundary`. 
Besides collecting some data for market purposes `trackEvent` is not essential for the end-users, we are always consuming the `trackEvent` along with a user's actions:
```jsx
<Button onClick={() => {
  trackEvent(...);
  onClickHandler();
}}>
``` 
In the example above if the `trackEvent` fails for some reason the user is actually affected.
So I would suggest changing the original implementation in order to ignore errors thrown from mixpanel as:
```ts
const evaluateTracking = (...args) => {
  // pseudo parallelism
  setTimeout(() => {
    try {
      const mixpanelPublicToken = process.env.MIXPANEL_PUBLIC_TOKEN;
      if (!mixpanelPublicToken || storage.local.read<boolean>(ANALYTICS_CONSENT_STORAGE_KEY) !== true) {
        return;
      }
      mx.init(mixpanelPublicToken);
      mx.track(...args);
    } catch(ignore) {
      // ignored
    }
  }, 0)
};
```

---

Instead of setTimeout, you can use `window.requestIdleCallback`

---

Very nice thought, to safeguard this. Implemented!

---

perhaps wrap the "try-catch" block in a `window.requestIdleCallback(() => { ... })`. This makes the browser execute this only when it has "idle time".

Unfortunately, Safari doesn't have support for this, so in our utils we have to add

```ts
window.requestIdleCallback = window.requestIdleCallback || (cb => cb());
```

and update `web/src/helpers/fireworks` to use `window.requestIdleCallback` since currently it uses a local version of what I wrote above (it re-defines it)